exchanges/poloniex/poloniex_types.go

```diff
@@ -446,75 +125,906 @@ var WithdrawalFees = map[currency.Code]float64{
 }
 
-// WalletActivityResponse holds wallet activity info
-type WalletActivityResponse struct {
-	Deposits    []WalletDeposits    `json:"deposits"`
-	Withdrawals []WalletWithdrawals `json:"withdrawals"`
+// WsCancelOrderResponse represents a websocket cancel orders instance.
+type WsCancelOrderResponse struct {
+	OrderID int64 `json:"orderId"`
+	CancelOrderResponse
 }
 
-// WalletDeposits holds wallet deposit info
-type WalletDeposits struct {
-	DepositNumber int64         `json:"depositNumber"`
-	Currency      currency.Code `json:"currency"`
-	Address       string        `json:"address"`
-	Amount        float64       `json:"amount,string"`
-	Confirmations int64         `json:"confirmations"`
-	TransactionID string        `json:"txid"`
-	Timestamp     types.Time    `json:"timestamp"`
-	Status        string        `json:"status"`
+// CancelOrdersRequest represents cancel order request parameters
+type CancelOrdersRequest struct {
+	Symbol         currency.Pair `json:"symbol"`
+	OrderIDs       []string      `json:"ordIds,omitempty"`
+	ClientOrderIDs []string      `json:"clOrdIds,omitempty"`
 }
 
-// WalletWithdrawals holds wallet withdrawal info
-type WalletWithdrawals struct {
-	WithdrawalRequestsID int64         `json:"withdrawalRequestsId"`
-	Currency             currency.Code `json:"currency"`
-	Address              string        `json:"address"`
-	Amount               float64       `json:"amount,string"`
-	Fee                  float64       `json:"fee,string"`
-	Timestamp            types.Time    `json:"timestamp"`
-	Status               string        `json:"status"`
-	TransactionID        string        `json:"txid"`
-	IPAddress            string        `json:"ipAddress"`
-	PaymentID            string        `json:"paymentID"`
-}
-
-// TimeStampResponse returns the time
-type TimeStampResponse struct {
-	ServerTime types.Time `json:"serverTime"`
+// KillSwitchStatus represents a kill switch response
+type KillSwitchStatus struct {
+	StartTime        types.Time `json:"startTime"`
+	CancellationTime types.Time `json:"cancellationTime"`
+}
+
+// SmartOrderRequestRequest represents a smart trade order parameters
+type SmartOrderRequestRequest struct {
+	Symbol         currency.Pair `json:"symbol"`
+	Side           order.Side    `json:"side"`
+	TimeInForce    timeInForce   `json:"timeInForce,omitempty"`
+	AccountType    accountType   `json:"accountType,omitempty"`
+	Type           string        `json:"type,omitempty"`
+	Price          float64       `json:"price,omitempty,string"`
+	StopPrice      float64       `json:"stopPrice,omitempty,string"`
+	Quantity       float64       `json:"quantity,omitempty,string"`
+	Amount         float64       `json:"amount,omitempty,string"`
+	ClientOrderID  string        `json:"clientOrderId,omitempty"`
+	TrailingOffset string        `json:"trailingOffset,omitempty"` // trailing stop price offset (Suffix with%, to trailing the proportion, without%, to trailing the price distance)
```

Comment:
🚧 Change request:
I see this came verbatim from their docs. I'm still going to simplify it massively
```suggestion
	TrailingOffset string        `json:"trailingOffset,omitempty"` // trailing stop offset; Append % to trail by percentage
```